transformers4rec/torch/model/base.py

```diff
@@ -678,6 +697,143 @@ def _get_name(self):
 
         return super(Model, self)._get_name()
 
+    def input_schema(self, max_sequence_length=None):
+        # return the input schema given by the model
+        # loop over the heads to get input schemas
+        schemas = []
+        for head in self.heads:
+            schemas.append(head.body.inputs.schema)
+        model_schema = sum(schemas, Schema())
+
+        # TODO: rework T4R to use Merlin Schemas.
+        # In the meantime, we convert model_schema to merlin core schema
+        core_schema = Core_Schema()
+        for column in model_schema:
+            name = column.name
+
+            dtype = {0: np.float32, 2: np.int64, 3: np.float32}[column.type]
+            tags = column.tags
+            is_list = column.value_count.max > 0
+            # T4Rec is only supporting padded dense input tensors with a fixed max_sequence_length
+            if is_list and not max_sequence_length:
+                raise ValueError(
+                    f"You should specify the `max_sequence_length` of your list input {name}"
+                )
+            value_count = (
+                {"min": max_sequence_length, "max": max_sequence_length}
+                if is_list
+                else {"min": 1, "max": 1}
+            )
+            is_ragged = is_list and value_count.get("min", 0) != value_count.get("max", 0)
+            int_domain = {"min": column.int_domain.min, "max": column.int_domain.max}
+            shape = max_sequence_length if is_list else 1
+            properties = {"value_count": value_count, "int_domain": int_domain, "shape": shape}
```

Comment:
In Merlin Systems, this `shape` is expected to include the batch dimension specified as -1 e.g. `[-1, 20]` for a list of size 20 or `[-1]` for a single value.

Perhaps we need to agree on the convention we're going for with this shape or maybe pull this shape into the Schema so that this can be checked and made consistent

Reply:
Thank you @oliverholworthy for your review! Indeed adding the shape information to the schema properties was suggested by Karl when we were discussing what would be the best option to forward the information of non-ragged list features to merlin-systems. But sorry, I missed the standard format used in systems...
The shape information is specific to systems.  So I believe we can update the in/out schema in T4Rec to follow the same convention used in systems. 
Just to double-check, the shape of features is set in this [compute_dim](https://github.com/NVIDIA-Merlin/systems/blob/15074adf11bcc9bb9da375ec40ab4e431f1c93e9/merlin/systems/dag/ops/__init__.py#L18) function? 